### ISSUE TITLE
Accuracy in multi-task scenarios

### ISSUE THREAD

**Issue by AntonioCarta** [ContinualAI/avalanche#1087]
I'm doing some experiments with multi-task scenarios. Right now Avalanche splits accuracy by task, but I find this a bit inconvenient. For example, I would like to have a single stream accuracy (averaged over tasks) but this is not what Avalanche gives. Notice that in the common scenario (different task for each exp) the ExpAccuracy already gives the task-accuracy.

I think the ideal solution would be to have separate task-aware and task-agnostic metrics, with the default being task-agnostic. A single metric plugin could even track both, or recognize automatically whether the benchmark is task-aware/task-agnostic. Notice that this would be a breaking change since metric names will change slightly.

@AndreaCossu what do you think?

**Comment by AndreaCossu** [comment]
I agree. We already encountered the problem you mentioned with @vlomonaco . An easy fix would be, whenever the stream accuracy metric is defined, to always return the average across all experiences, since that is usually what the user was looking for. In most cases, task-aware settings are used just for training and not for evaluation.
The `EvaluationPlugin` can recognize if the benchmark is task-agnostic if the user provides the benchmark object to it, otherwise it may be cumbersome to do so in a general way.

**Comment by AntonioCarta** [comment]
Ok, I will work on this since I need it now ;)
